src/router/db.ts

```diff
@@ -229,6 +238,72 @@ db.post('/rpc', async (req: Request, res: Response) => {
   }
 });
 
+async function rpcCall(
+  dbId: string,
+  dbUser: string,
+  moduleInstanceName: string,
+  methodname: string,
+  params: string[],
+  context: Context,
+  conn: TypeormWrapper,
+  preTransaction?: PreTransactionCheck,
+  postTransaction?: PostTransactionCheck,
+): Promise<any[] | undefined> {
+  let rpcRes: any[] | undefined;
+  let errorOcurred = false;
+  let finallyErr;
+  try {
+    switch (preTransaction) {
+      case PreTransactionCheck.NO_CHECK:
+        break;
+      case PreTransactionCheck.FAIL_IF_NOT_LOCKED:
+        const openTransaction = await iasqlFunctions.isOpenTransaction(conn);
+        if (!openTransaction) {
+          throw new Error('Cannot execute without calling iasql_begin() first');
+        }
+        break;
+      case PreTransactionCheck.WAIT_FOR_LOCK:
+        await iasqlFunctions.maybeOpenTransaction(conn);
+        break;
+      default:
+        await conn.query(`select * from iasql_begin();`);
```

Comment:
What is this for? Is there a case we'll need it?

Reply:
It is the default behaviour for cases like the one Yolanda has where the rpc is not doing the whole heavy load of syncing the DB. With the checks we only open and close transactions but never commit at the end.

Reply:
Oh got it. So this is just for the transactions and not for `commit`/`sync`, right?

Reply:
not sure I follow your question